File: scratch/FizzBuzz/FizzBuzz.scala

def toFizzBuzz(n: Int) = {
    if (n.toString contains "3") "lucky"
    else if (n % 15 == 0) "fizzbuzz"
    else if (n % 3 == 0) "fizz"
    else if (n % 5 == 0) "buzz"
    else n
}                                             

val start = System.currentTimeMillis  
val fizzBuzzed = (1 to 20) map toFizzBuzz
println(fizzBuzzed)
val stop = System.currentTimeMillis
val time = stop - start
println(s"Time Taken (ms) = $time")

File: scratch/FizzBuzz/FizzBuzz.hs
import Data.List (isInfixOf)
import Data.Time (getCurrentTime, diffUTCTime)

fizzBuzz :: Int -> String
fizzBuzz n
  | n `mod` 15 == 0 = "FizzBuzz"
  | n `mod` 3 == 0 = "Fizz"
  | n `mod` 5 == 0 = "Buzz"
  | otherwise =  ""

-- Part 1
-- fizzBuzz :: Int -> String
-- fizzBuzz n
--   | n `mod` 15 == 0 = "FizzBuzz"
--   | n `mod` 3 == 0 = "Fizz"
--   | n `mod` 5 == 0 = "Buzz"
--   | otherwise = show n
  
-- Part 2
-- fizzBuzz :: Int -> String
-- fizzBuzz n
--   | isInfixOf "3" (show n) = "Lucky"
--   | n `mod` 15 == 0 = "FizzBuzz"
--   | n `mod` 3 == 0 = "Fizz"
--   | n `mod` 5 == 0 = "Buzz"
--   | otherwise = show n

main :: IO ()
main = do  
  start <- getCurrentTime
  print $ map fizzBuzz [1..2000]
  stop <- getCurrentTime
  print $ diffUTCTime stop start
  print "DONE"